### PR TITLE
Added: Functions to delete extended data columns

### PR DESCRIPTION
**Pull request by joeferner**

### CODE REVIEW

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraph.java [hunk 1]
```diff
@@ -338,6 +349,26 @@ public Vertex save(Authorizations authorizations) {
                         for (PropertyDeleteMutation propertyDeleteMutation : getPropertyDeletes()) {
                             queueEvent(new DeletePropertyEvent(AccumuloGraph.this, vertex, propertyDeleteMutation));
                         }
```
[the remark this answers is not in the document]
Comment (joeferner) [reply accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraph.java]: They should. Although we don't use any of the events so I'm wondering if all the event stuff should all be ripped out.

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraph.java [hunk 2]
```diff
@@ -1019,13 +1081,40 @@ public AccumuloNameSubstitutionStrategy getNameSubstitutionStrategy() {
...
             return;
         }
 
-        elementMutationBuilder.saveExtendedDataMarkers(elementId, elementType, extendedData);
-        elementMutationBuilder.saveExtendedData(this, elementId, elementType, extendedData);
+        elementMutationBuilder.saveExtendedDataMarkers(element.getId(), elementType, extendedData);
+        elementMutationBuilder.saveExtendedData(this, element.getId(), elementType, extendedData);
+        elementMutationBuilder.saveExtendedDataDeletes(this, element.getId(), elementType, extendedDataDeletes);
+
+        for (ExtendedDataMutation extendedDataMutation : extendedData) {
```
[the remark this answers is not in the document]
Comment (joeferner) [reply accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraph.java]: yep, I can fix this

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraph.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (joeferner) [reply accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraph.java]: done

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraph.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (joeferner) [reply accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraph.java]: done

**Comment by joeferner** [comment]
@mwizeman Looks like I have a real bug. I'll take a look tonight

**Comment by mwizeman** [comment]
@joeferner I'm running those tests locally to verify that the failure isn't travis related.

**Comment by joeferner** [comment]
I'm wondering if I missed a delete from search index somewhere in the `AccumuloGraph` class which could cause that behavior or returning 0 rows but saying there is 1 hit.

**Comment by mwizeman** [comment]
@joeferner Had the same failure locally. I can take a quick look.

**Pull request merged**